packages/salesforcedx-vscode-core/src/commands/templates/forceApexClassCreate.ts

```diff
@@ -43,7 +48,9 @@ export class LibraryForceApexClassCreateExecutor extends LibraryBaseTemplateComm
   }
 }
 
-const fileNameGatherer = new SelectFileName();
+const fileNameGatherer = new SelectFileName(
+  new MaxLengthValidator(APEX_CLASS_NAME_MAX_LENGTH)
```

Comment:
validators are injected and optional. This way other instances of `SelectFileName` won't break